rust/connlib/clients/shared/src/control.rs

```diff
@@ -229,8 +229,10 @@ impl<CB: Callbacks + 'static> ControlPlane<CB> {
                 };
 
                 tokio::spawn(async move {
-                    if let Err(e) = upload(path, url).await {
+                    if let Err(e) = upload(path.clone(), url).await {
```

Comment:
Is this `clone()` ok here?

Reply:
Need because path is being borrowed here

Reply:
Seems to work great -- log file is uploaded, deleted and new one immediately started.

Reply:
> Is this `clone()` ok here?

Yeah sure!

Reply:
[nit] you can change `upload` to take a `&Path` instead of `PathBuf` and you wouldn't need to clone :p